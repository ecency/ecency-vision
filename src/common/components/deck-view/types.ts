export interface DeckModel {
  data: any;
  dataParams: any[];
  header: {
    title: string;
    icon: JSX.Element | null;
  };
  listItemComponent: any;
  createdAt: Date;
}

export interface IdentifiableDeckModel extends DeckModel {
  id: string;
  content: string;
}
