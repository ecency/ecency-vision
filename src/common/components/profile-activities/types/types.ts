export interface ActivityTypes {
    type: string;
      parent_author?: string;
      parent_permlink?: string;
      author?: string;
      permlink?: string;
      payment?: number;
      payer?: string;
      following?: string;
      community?: string;
      approve?: boolean;
      proposal_ids?: string;
      account?: string;
      id?: string;
      what?: string[];
      voter?: string;
      witness?: string;
      json: string;
    timestamp: string;
  }