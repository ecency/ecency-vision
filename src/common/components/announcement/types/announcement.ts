export interface Announcement {
  id: number;
  title: string;
  description: string;
  button_text: string;
  button_link: string;
}

export interface LaterAnnouncement {
  id: number;
  dateTime: Date;
}
