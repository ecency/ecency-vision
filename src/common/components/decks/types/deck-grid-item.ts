interface BaseDeckGridItem {
  id: string;
  key: number;
  type:
    | "u"
    | "ac"
    | "co"
    | "w"
    | "n"
    | "tr"
    | "to"
    | "s"
    | "cu"
    | "th"
    | "msf"
    | "faq"
    | "wb"
    | "wn";
  settings: Record<string, any>;
}

export interface ReloadableDeckGridItem extends BaseDeckGridItem {
  settings: {
    updateIntervalMs: number;
  };
}

export interface UserDeckGridItem extends ReloadableDeckGridItem {
  type: "u";
  settings: ReloadableDeckGridItem["settings"] & {
    username: string;
    contentType: string;
  };
}

export interface CommunityDeckGridItem extends ReloadableDeckGridItem {
  type: "co";
  settings: UserDeckGridItem["settings"] & {
    tag: string;
  };
}

export interface SearchDeckGridItem extends ReloadableDeckGridItem {
  type: "s";
  settings: ReloadableDeckGridItem["settings"] & {
    query: string;
    author?: string;
    type?: string;
    category?: string;
    tags?: string;
    date?: string;
    sort?: string;
    hideLow?: boolean;
  };
}

export interface WavesDeckGridItem extends ReloadableDeckGridItem {
  type: "th";
  settings: ReloadableDeckGridItem["settings"] & {
    host: string;
  };
}

export type DeckGridItem =
  | BaseDeckGridItem
  | UserDeckGridItem
  | CommunityDeckGridItem
  | SearchDeckGridItem
  | WavesDeckGridItem;
