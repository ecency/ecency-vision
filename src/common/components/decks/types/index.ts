export * from "./deck-grid-item";
export * from "./deck-grid";
export * from "./old-decks";
