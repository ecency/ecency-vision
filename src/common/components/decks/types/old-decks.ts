interface GenericDeck<DF extends Record<string, string | number> | null> {
  createdAt: string;
  dataFilters: DF;
  dataParams: string[];
  header: {
    title: string;
    updateIntervalMs: number;
  };
}

export interface OldNotificationDeck extends GenericDeck<{ type: string; group: string }> {}
export interface OldNonDFDeck extends GenericDeck<null> {}

export type OldDeck = OldNotificationDeck | OldNonDFDeck;
