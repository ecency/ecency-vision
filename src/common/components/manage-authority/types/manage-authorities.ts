import { Account, FullAccount } from "../../../store/accounts/types";
import { PublicKey } from "@hiveio/dhive";

export const Keytype = {
  Owner: "owner",
  Active: "active",
  Posting: "posting",
  Memo: "memo"
};

export interface PublicKeys {
  publicOwnerKey?: string;
  publicActiveKey?: string;
  publicPostingKey?: string;
  publicMemoKey?: string;
}

export interface AccountDataType {
  postingsAuthority: [string, number][];
  posting: [string | PublicKey, number];
  owner: [string | PublicKey, number];
  active: [string | PublicKey, number];
  weight: number;
  memokey: string;
  account: Account;
  publicKeys: {
    publicOwnerKey: string | PublicKey;
    publicActiveKey: string | PublicKey;
    publicPostingKey: string | PublicKey;
    publicMemoKey: string;
  };
}

export const actionType = {
  Revoke: "revoke",
  Keys: "keys",
  Import: "import",
  Reveal: "reveal"
};
