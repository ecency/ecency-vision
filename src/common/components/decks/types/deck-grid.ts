import { DeckGridItem } from "./deck-grid-item";

export interface DeckGrid {
  key: string;
  title: string;
  icon?: string;
  columns: DeckGridItem[];
  storageType: "account" | "local";
}

export interface DeckGrids {
  decks: DeckGrid[];
}
