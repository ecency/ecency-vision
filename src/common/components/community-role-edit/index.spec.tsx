import React from "react";

import renderer from "react-test-renderer";

import {CommunityRoleEdit} from "./index";

import {roleMap} from "../../store/communities/types";

import {communityInstance1, globalInstance, activeUserMaker} from "../../helper/test-helper";

it("(1) Render", () => {
    const props = {
        global: globalInstance,
        community: {...communityInstance1},
        activeUser: activeUserMaker("hive-148441"),
        user: 'foo',
        role: 'mod',
        roles: roleMap["owner"],
        addCommunity: () => {
        },
        onHide: () => {
        }
    };

    const component = renderer.create(<CommunityRoleEdit {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
});
