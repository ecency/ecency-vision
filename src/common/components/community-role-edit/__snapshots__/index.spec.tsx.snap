// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(1) Render 1`] = `
<div
  className="community-role-edit-dialog-content"
>
  <div
    className="user-role-form "
  >
    <div
      className="flex mb-4"
    >
      <div
        className="w-full sm:w-2/12"
      >
        <label>
          Username
        </label>
      </div>
      <div
        className="w-full sm:w-10/12"
      >
        <div
          className="ecency-input-group flex items-stretch w-full [&>input]:rounded-[0] [&>input:first-child]:rounded-l-full [&>input:last-child]:rounded-r-full [&>.ecency-input-group-part>button]:rounded-tl-none [&>.ecency-input-group-part>button]:rounded-bl-none [&>.ecency-input-group-part>button]:h-[2.75rem]"
        >
          <div
            className="ecency-input-group-part ecency-input-group-prepend flex items-center justify-center border-r-0 rounded-tl-full rounded-bl-full bg-gray-200 dark:bg-gray-600 dark:border-gray-600 px-2.5 [&>svg]:w-4 [&>svg]:h-4 px-2 border-2"
            onClick={[Function]}
          >
            @
          </div>
          <input
            autoFocus={false}
            className="border-2 rounded-3xl py-2 px-3 w-full outline-none shadow-0 focus:border-gray-500 hover:border-gray-300 duration-300 dark:border-gray-700 dark:hover:border-gray-600 dark:bg-gray-800 aria-invalid:border-red"
            onChange={[Function]}
            placeholder="username"
            type="text"
            value="foo"
          />
        </div>
        
      </div>
    </div>
    <div
      className="flex mb-4"
    >
      <div
        className="w-full sm:w-2/12"
      >
        <label>
          Role
        </label>
      </div>
      <div
        className="w-full sm:w-10/12"
      >
        <select
          className="border-2 rounded-3xl py-2 px-3 w-full outline-none shadow-0 focus:border-gray-500 hover:border-gray-300 duration-300 dark:border-gray-700 dark:hover:border-gray-600 dark:bg-gray-800 "
          onChange={[Function]}
          type="select"
          value="mod"
        >
          <option
            value="admin"
          >
            admin
          </option>
          <option
            value="mod"
          >
            mod
          </option>
          <option
            value="member"
          >
            member
          </option>
          <option
            value="guest"
          >
            guest
          </option>
          <option
            value="muted"
          >
            muted
          </option>
        </select>
      </div>
    </div>
    <div
      className="flex justify-end"
    >
      <button
        className="cursor-pointer rounded-full duration-300 no-wrap bg-blue-dark-sky hover:bg-blue-dark-sky-hover focus:bg-blue-dark-sky-active text-white disabled:opacity-50 disabled:hover:bg-blue-dark-sky disabled:focus:bg-blue-dark-sky h-[2.125rem] px-3 "
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "outline": "none",
          }
        }
        type="button"
      >
        <div>
          Save
        </div>
      </button>
    </div>
  </div>
  <div
    className="explanations"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<h3>Role Permissions</h3><p><strong>Owner</strong> - assign admins</p><p><strong>Admin</strong> - edit settings, assign mods</p><p><strong>Moderator</strong> - mute, pin, set user titles</p><p><strong>Member</strong> - listed on leadership team</p><p><strong>Guest</strong> - default; can post and comment</p><p><strong>Muted</strong> - new posts automatically muted</p>",
      }
    }
  />
</div>
`;
